### PR TITLE
changelog: improve description of smart dialing

### PR DESCRIPTION
**Pull request by marten-seemann**

### CODE REVIEW

**Review by sukunrt — APPROVED** [review]

**Comment by marten-seemann** [comment]
> Do you want to mention the two metrics for tracking smart dialing impact?
> `dial ranking delay and dials per connection`

Done.

**Review by sukunrt — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,15 +14,38 @@
...
+on the server side, and is heavily discouraged by the networking community (see [RFC 8305](https://www.rfc-editor.org/rfc/rfc8305) for example).
+
+When connecting to a peer we first determine the order to dial its addresses. This ranking logic considers a number of corner cases
+described in detail in the documentation of the swarm package (`swarm.DefaultDialRanker`).
+At a high level, this is what happens:
+* If a peer offers a WebTransport and a QUIC address (on the same IP:port), the QUIC address is preferred.
+* If a peer has a QUIC and a TCP address, the QUIC address is dialed first. Only if the connection attempt doesn't succeed within 250ms, a TCP connection is started.
+
+Our measurements on the IPFS network show that for >90% of established libp2p connections, the first connection attempt succeeds,
+leading a dramatic decrease in the number of aborted connection attempts.
```
Comment (BigLep) [inline CHANGELOG.md:30]: Can we put numbers to this to inform how dramatic this was?  For example, I think the "Cancel fraction" we had in https://github.com/libp2p/go-libp2p/pull/2260#issuecomment-1528848170 was useful.  

I'm thinking about what we'd say in a presentation or all hands.  Is this reducing the number of "wasted" connections by 50%, 75%, etc.

**Review by BigLep — COMMENTED** [review]
I have a suggestion about being a bit more precise so someone can get a hold of the impact, but otherwise looks great - thanks.

File: CHANGELOG.md (hunk 1 above)
Comment (sukunrt) [reply CHANGELOG.md:30]: I ran the numbers again for master branch. This is reducing the number of "wasted" connections by 50%.
With no smart dialing 60% of all connection attempts are cancelled. With smart dialing turned on 30% of all connection attempts are cancelled.

Comment (marten-seemann) [reply CHANGELOG.md:30]: We can report numbers here, or on the next release when we have black hole detection (or both). I expect the numbers to be significantly better when we have that, as there are quite a few dual stack hosts in the network.

**Review by MarcoPolo — APPROVED** [review]

**Comment by mxinden** [comment]
I like the detailed changelog entry! Very helpful as an outsider.

**Pull request merged**